### PR TITLE
feat: poc: Use events for updates

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

### CODE REVIEW

File: server/workflow/workflow_server.go [hunk 1]
```diff
@@ -242,39 +242,27 @@ func (s *workflowServer) ResubmitWorkflow(ctx context.Context, req *workflowpkg.
...
-
-	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), s.offloadNodeStatusRepo, req.Name, req.NodeFieldSelector)
+	wf, err := s.getWorkflowAndValidate(ctx, req.Namespace, req.Name, metav1.GetOptions{})
 	if err != nil {
-		log.Warnf("Failed to resume %s: %+v", req.Name, err)
 		return nil, err
 	}
 
-	wf, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Get(req.Name, metav1.GetOptions{})
+	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().WorkflowOps(req.Namespace), wf, req.NodeFieldSelector)
```
Comment (alexec) [inline server/workflow/workflow_server.go:250]: challenge one: now decoupled - we cannot determine if the op was valid

File: workflow/controller/controller.go [hunk 2]
```diff
@@ -424,13 +439,21 @@ func (wfc *WorkflowController) runWorker() {
 
 // processNextItem is the worker logic for handling workflow updates
 func (wfc *WorkflowController) processNextItem() bool {
-	key, quit := wfc.wfQueue.Get()
+	item, quit := wfc.wfQueue.Get()
```
Comment (alexec) [inline workflow/controller/controller.go:442]: queuing different things works well

File: workflow/controller/operator.go [hunk 3]
```diff
@@ -2618,3 +2620,51 @@ func (woc *wfOperationCtx) includeScriptOutput(nodeName, boundaryID string) (boo
 	}
 	return false, nil
 }
+
+func (woc *wfOperationCtx) executeWorkflowOp(op wfv1.WorkflowOp) {
+	if op.Spec.Suspend != nil {
+		woc.log.Info("Suspending workflow")
```
Comment (alexec) [inline workflow/controller/operator.go:2626]: code migrates from util

File: workflow/util/util_test.go [hunk 4]
```diff
@@ -279,91 +279,26 @@ status:
...
-
-	clearFunc := packer.SetMaxWorkflowSize(1156)
-	defer clearFunc()
-	err := packer.CompressWorkflowIfNeeded(origWf)
-	assert.NoError(t, err)
-
-	_, err = wfIf.Create(origWf)
-	assert.NoError(t, err)
-
-	err = ResumeWorkflow(wfIf, sqldb.ExplosiveOffloadNodeStatusRepo, "suspend", "")
```
Comment (alexec) [inline workflow/util/util_test.go:294]: all the repo nastiness is removed

File: server/workflow/workflow_server.go (hunk 1 above)
Comment (mark9white) [reply server/workflow/workflow_server.go:254]: This is a potential issue. At the moment my client code retries the 'resume' a few times on failure, as it's possible the resume node hasn't been created yet.

**Comment by alexec** [comment]
Abandoning this PR - too invasive changes. Will try and fix in refactor of code instead.

**Pull request closed without merging**